src/main/java/com/exasol/adapter/dynamodb/documentpath/RedundantPathEliminator.java

```diff
@@ -0,0 +1,91 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class eliminates path expressions that are redundant when used as a projection expression.
+ *
+ * For example {@code a} and {@code a.b} are reduced to {@code a} as {@code a} includes {@code a.b}.
+ */
+public class RedundantPathEliminator {
+    private static final RedundantPathEliminator INSTANCE = new RedundantPathEliminator();
+
+    /**
+     * Empty constructor to hide public default.
+     */
+    private RedundantPathEliminator() {
+        // empty on purpose.
+    }
+
+    /**
+     * Get a singleton instance of {@link RedundantPathEliminator}.
+     *
+     * @return instance of {@link RedundantPathEliminator}
+     */
+    public static RedundantPathEliminator getInstance() {
+        return INSTANCE;
+    }
+
+    /**
+     * Eliminate path expressions that are redundant when used as a projection expression.
+     *
+     * @param paths collection of paths
+     * @return Set with redundancy free paths
+     */
+    public Set<DocumentPathExpression> removeRedundantPaths(final Collection<DocumentPathExpression> paths) {
+        return removeRedundantPaths(paths.stream());
+    }
+
+    /**
+     * Eliminate path expressions that are redundant when used as a projection expression.
+     * 
+     * @param paths collection of paths
+     * @return Set with redundancy free paths
+     */
+    public Set<DocumentPathExpression> removeRedundantPaths(final Stream<DocumentPathExpression> paths) {
+        /*
+         * The basic idea of this algorithm is to iterate over the paths with an increasing path length. In the first
+         * iteration only the paths with a length of 0 are considered. In each iteration the algorithm removes all paths
+         * that are already included in the result from allPaths.
+         */
```

Comment:
Could you please add this to Javadoc (with implementation note tag)?.

Reply:
like that?